napari_skimage_regionprops/_utilities.py

```diff
@@ -1,18 +1,24 @@
 import numpy as np
-from napari_plugin_engine import napari_hook_implementation
-from napari.types import ImageData
-from napari import Viewer
+try:
+    from napari_plugin_engine import napari_hook_implementation
+    from napari.types import ImageData
+    from napari import Viewer
+    from napari.layers import Image, Labels, Layer
+    from typing_extensions import Annotated
+    LayerInput = Annotated[Layer, {"label": "Image"}]
+except ModuleNotFoundError as e:
+    import warnings
+    warnings.warn(str(e))
+
 from napari_tools_menu import register_function
-from typing_extensions import Annotated
-from napari.layers import Image, Labels, Layer
-LayerInput = Annotated[Layer, {"label": "Image"}]
+
 
 @napari_hook_implementation
 def napari_experimental_provide_function():
     return [duplicate_current_frame]
 
 @register_function(menu="Utilities > Duplicate current timepoint (nsr)")
-def duplicate_current_frame(layer : LayerInput, napari_viewer : Viewer, axis : int = 0) -> Layer:
+def duplicate_current_frame(layer : "LayerInput", napari_viewer: "Viewer", axis : int = 0) -> "Layer":
```

Comment:
Hi @tcompa ,

just to make you awara: This line made my napari crash when clicking on the menu. I had to modify it to this:
```
def duplicate_current_frame(layer : LayerInput, napari_viewer: "napari.Viewer", axis : int = 0) -> "napari.layers.Layer":
```

I'd say its necessary to click all menus in napari to check if the functionality still works. If you find a way for automatically testing those things, feedback is very welcome!

Thanks again :-)

Reply:
Thanks for checking!

To be honest, my goal was to use these functions outside napari, and I did not verify that they were still working fine in there. My bad.

I have no special experience with testing the GUI menu-clicking. Would https://pytest-qt.readthedocs.io/en/latest/tutorial.html?highlight=clicking#tutorial be an option?

Unrelated: unfortunately a first test suggests that there are still glitches in how imports are handled (either here or in napari-tools-menu), in case one wants to install this package without QT (briefly: create fresh conda env, `pip install napari_skimage_regionprops=0.8`, open python, `import napari_skimage_regionprops` fails). I will debug further and open an appropriate issue as soon as possible. Apologies for the required effort, I can appreciate that supporting two different use cases (GUI or headless) may indeed be tricky.